src/vim-webgl-viewer/settings/viewerSettingsParsing.ts

```diff
@@ -0,0 +1,163 @@
+import * as THREE from 'three'
+import { GizmoOptions } from '../gizmos/gizmoAxes'
+import { PartialViewerSettings , ViewerSettings } from './viewerSettings'
+import deepmerge from 'deepmerge';
+
+/**
+ * Parses the url for settings and merges the result with the optional provided settings.
+ * URL settings have priority
+ * @param settings Settings to use if not defined in the url. 
+ * @param url URL string with standard parameters.
+ * @returns A PartialSettings object that can further be merged with default values.
+ */
+export function getViewerSettingsFromUrl(url?: string, settings?: PartialViewerSettings){
+  const urlSettings = parseSettingsFromUrl(url)
+  return deepmerge(settings, urlSettings) as PartialViewerSettings
+}
+
+function parseSettingsFromUrl(url: string){
+  
+  const params = new URLSearchParams(url)
+  function get<T>(key: string, parse? :(str: string) => T){
```

Comment:
Is there a way to improve the type safety and maintainability of the possible string inputs of this function by using some TypeScript typing mechanism like [keyof] ?